catalyst/runners/hydra_supervised.py

```diff
@@ -0,0 +1,195 @@
+from typing import Any, Callable, List, Mapping, Tuple, Union
+import logging
+
+import torch
+
+from catalyst.experiments.hydra_config import HydraConfigExperiment
+from catalyst.runners.runner import Runner
+from catalyst.typing import Device, RunnerModel
+
+logger = logging.getLogger(__name__)
+
+
+class HydraSupervisedRunner(Runner):
```

Comment:
do we really need it?
from pipeline perspective (model & data flow)  - there are no changes, is't it?
from Config API - we could redefine used experiment with
```
experiment_params:
  ...
```
like https://github.com/catalyst-team/catalyst/blob/master/examples/configs/config-description-eng.yml#L26, but `experiment_params` :joy:

Reply:
another option - `HydraSupervisedRunner(SupervisedRunner)`

Reply:
Generally `HydraSupervisedRunner` is very different from `SupervisedRunner`.
1. __init__ parameters: `model_keys` except `input_key/output_key/input_target_key`
2. Multi-model support: there are separate `input_key/output_key/input_target_key` for each `model`
3. _process_input/_process_output/_handle_device/forward methods are different
But think I can inherit `HydraSupervisedRunner` from `SupervisedRunner`.
I wanted to minimize dependency between `HydraSupervisedRunner` and `SupervisedRunner`. But if you say so I rewrite.

Reply:
let's rename it to `MultiSupervisedRunner`? 

I also think that we should use `AutoCallbackExperiment` as the default.
`experiment_fn: Callable` is good for notebook experiments, but with Config API you could easily rewrite the default Experiment with 
```
experiment_params:
    ...
```
or just write it into `__init__.py`